client/data/media/use-update-media-mutation.js

```diff
@@ -0,0 +1,18 @@
+import { useCallback } from 'react';
+import { useMutation } from 'react-query';
+import wp from 'calypso/lib/wp';
+
+export const useUpdateMediaMutation = ( siteId, queryOptions = {} ) => {
+	const mutation = useMutation(
+		( { mediaId, updates } ) => wp.req.post( `/sites/${ siteId }/media/${ mediaId }`, updates ),
```

Comment:
There's a major difference to how this was done before. We used to always send the whole media object (including any attributes). By testing and looking at the backend side of this, however, this doesn't seem to be necessary.